src/core/net/netif.cpp

```diff
@@ -209,37 +218,36 @@ ThreadError Netif::UnsubscribeMulticast(const NetifMulticastAddress &aAddress)
 ThreadError Netif::SubscribeExternalMulticast(const Address &aAddress)
 {
     ThreadError error = kThreadError_None;
-    int8_t index = 0;
+    NetifMulticastAddress *entry;
+    size_t num = sizeof(mExtMulticastAddresses) / sizeof(mExtMulticastAddresses[0]);
```

Comment:
nit: Is there any reason not to use just OPENTHREAD_CONFIG_MAX_EXT_MULTICAST_IP_ADDRS explicitly? (same for other parts of code).